src/complex/Utilities/OStreamUtilities.hpp

```diff
@@ -0,0 +1,765 @@
+#pragma once
+
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/DataObject.hpp"
+#include "complex/DataStructure/DataPath.hpp"
+#include "complex/DataStructure/DataStructure.hpp"
+#include "complex/DataStructure/StringArray.hpp"
+#include "complex/Filter/IFilter.hpp"
+#include "complex/Utilities/ParallelDataAlgorithm.hpp"
+#include "complex/Utilities/StringUtilities.hpp"
+
+#include <fstream>
+#include <ostream>
+#include <sstream>
+#include <vector>
+
+namespace complex
+{
+namespace OStreamUtilities
+{
+
+/**
+ * @brief This is the primary datastructure used in print string assembly. It is a wrapper struct for
+ * adding additionaly functionality/readability (underlying datastructure is a std::vector)
+ * @param numrows The number of rows for the matrix
+ * @param numcols The number of columns for the matrix
+ */
+struct PrintMatrix2D
```

Comment:
```suggestion
class PrintMatrix2D
```
Since this contains both public and private members, I'd change this to `class`. There is no difference in the code generated. It's a typical convention to use `struct` for objects that are all public.

Additionally is this part of the public API for these utilities? If not this and the associated classes below, could be moved into the cpp file.